modules/packages/LinearAlgebra.chpl

```diff
@@ -1225,15 +1299,17 @@ module Sparse {
     if this.domain._value.parentDom != A.domain._value.parentDom then
       halt('Cannot add sparse arrays with non-matching parent domains');
 
-    // Create a copy of 'this'
+    // Create copy of 'this'
     var BDom = this.domain;
     var B: [BDom] this.eltType;
-    forall (i,j) in B.domain do A[i,j] = this[i,j];
+    forall (i,j) in B.domain do B[i,j] = this[i,j];
 
-    // Bulk add new indices
-    BDom += A.domain;
+    // If domain indices do not match, bulk add A's indices to B
+    if this.domain != A.domain {
```

Comment:
Could/should this be a check in sparse's bulkAdd?

Anyways, no need to change for this PR.